massa-execution-worker/src/execution.rs

```diff
@@ -1260,15 +1381,49 @@ impl ExecutionState {
             self.mip_store.clone(),
         );
 
+        // Deferred calls
+        let calls = execution_context.deferred_calls_advance_slot(
+            slot.clone(),
+            DEFERRED_CALL_MAX_FUTURE_SLOTS,
+            self.config.thread_count,
+        );
+
         // Get asynchronous messages to execute
         let messages = execution_context.take_async_batch(
```

Comment:
let's move async execution AFTER block execution (see spec) so that we allow async message execution to use all the unused gas of the slot.

It would be like this:

1- execute deferred calls (and count how much gas they used)
2 - execute block if not miss, (and count how much gas it used)
3 - execute async messages and give them all the unused slot gas

Reply:
done